File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int arr[200100]={0};
        for (int i=0;i<numbers.size();i++)
        {
            arr[numbers[i]+1000]++;
        }
        
        vector<int>val;
        for (int i=0;i<numbers.size();i++)
        {
            int temp=target-numbers[i];
            if (arr[temp+1000]>1 && temp==numbers[i])
            {
                val.push_back(i+1);
            }
              if (arr[temp+1000]>0 && temp!=numbers[i])
            {
                val.push_back(i+1);
            }
        }
        
        return val;
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int arr[] = new int[2];
        int counter=0;
        int frequencyarray[]= new int [4010];
        
        for (int i=0;i<numbers.length;i++)
        {
         frequencyarray[numbers[i]+1000]++;
        }
      //  System.out.println(frequencyarray[1007]);
       
        for (int i=0;i<numbers.length;i++)
        {
            int temp=target-numbers[i];
            System.out.println(temp);
            
            if (frequencyarray[temp+1000]>0 && temp!=numbers[i])
            {
                arr[counter]=i+1;
                counter++;
            }
             if (frequencyarray[temp+1000]>1)
            {
                arr[counter]=i+1;
                counter++;
            }
        }
        return arr;
        
    }
}